### ISSUE TITLE
Dropdown: space key cannot be prevented due to event being internal to the host element.

### ISSUE THREAD

**Issue by driskull** [Esri/calcite-design-system#9750]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

Dropdown: space key cannot be prevented due to event being internal to the host element.

### Expected Behavior

If you preventdefault on the space key on the host element it shouldn't do anything when space bar is pressed

### Reproduction Sample

https://codepen.io/driskull/pen/eYwmpwJ

### Reproduction Steps

1. open the sample
2. tab to the dropdown
3. hit space key
4. space key shouldn't do anything since event was prevented

### Reproduction Version

2.9.0

### Relevant Info

https://community.esri.com/t5/calcite-design-system-questions/calcite-input-with-calcite-dropdown/m-p/1503079#M659

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

N/A

**Comment by moui** [comment]
Thanks for creating the issue @driskull. I reported this in the esri community. I've tried all sorts of things to work around it with no luck. I'm trying to solve this for an ongoing development. The calcite-input with suggestions/autocomplete is a very useful feature.

**Comment by driskull** [comment]
For phase 1, I think the only logical solution is to move the keydown handler to the host element instead of on the internal element? Thoughts? @macandcheese @benelan @jcfranco 

This will allow the end user a chance to cancel the event.

**Comment by driskull** [comment]
Actually, I think we can close this one. I can accomplish what @moui is requesting here:

https://codepen.io/driskull/pen/KwwPXMw?editors=1011

Also, I think this use case is better suited by the autocomplete component. So maybe we can close this one? @moui thoughts?

**Comment by moui** [comment]
Yes,you can go ahead and close this. It came up when the Autocomplete component didn't exist and we were trying to build our own.

Thanks.

**Comment by geospatialem** [comment]
Closing per https://github.com/Esri/calcite-design-system/issues/9750#issuecomment-2803178614 and https://github.com/Esri/calcite-design-system/issues/9750#issuecomment-2808243447, where the new `autocomplete` component covers the use case 💪🏻
